File: 2-GettingStartedWithImages/cpp/GettingStartedWithImages/GettingStartedWithImages.cpp
// GettingStartedWithImages.cpp : This file contains the 'main' function. Program execution begins and ends there.
//

#include <opencv2/core.hpp>
#include <opencv2/imgcodecs.hpp>
#include <opencv2/highgui.hpp>

#include <iostream>

using namespace cv;

int main()
{
    std::string image_path = samples::findFile("starry_night.jpg");

    // IMREAD_COLOR loads the image in the BGR 8 - bit format.This is the default that is used here.
    // IMREAD_UNCHANGED loads the image as is(including the alpha channel if present)
    // IMREAD_GRAYSCALE loads the image as an intensity one
    Mat img = imread(image_path, IMREAD_COLOR);
    if (img.empty())
    {
        std::cout << "Could not read the image: " << image_path << std::endl;
        return 1;
    }
    imshow("Display window", img);
    int k = waitKey(0); // Wait for a keystroke in the window
    if (k == 's')
    {
        imwrite("starry_night.png", img);
    }
    return 0;
}

File: 2-GettingStartedWithImages/py/getting_started_with_images.py
import cv2 as cv
import os
import sys


# 'OPENCV_DIR' refers to a vc++ build tool dir, such as 'C:\OpenCV\Build\x64\vc14'
sample_path = os.getenv('OPENCV_DIR') + "\\..\\..\\..\\sources\\samples\\data"
cv.samples.addSamplesDataSearchPath(sample_path)

img = cv.imread(cv.samples.findFile("starry_night.jpg"))

if (img is None):
    sys.exit("Could not read the image.")

cv.imshow("Display window", img)
k = cv.waitKey(0)

if (k == ord("s")):
    cv.imwrite("starry_night.png", img)